### ISSUE TITLE
Single-sided b2Edge / b2ChainShape.

### ISSUE THREAD

**Issue by mcourteaux** [erincatto/box2d#522]
Hey Erin,

A couple of years ago, I made some adjustments to the collision resolving to support - what I called - _single sided edges_. The default mechanism seeks the closest exit of the edge to resolve the collision. But for static scenery, it makes sense to make the edge have a default "outward direction" which will be used to resolve the collision. This reduces tunneling ***a lot*** (actually this was the best improvement I have done to the entire physics in my game; this was a huge win). The convention of what is "outward", simply follows the winding direction, like polygons. The single-sided feature was a `bool m_singleSided;` (default `false`) in `b2ChainShape` and `b2Edge`.

In short: 

 1. `b2CollideEdge.cpp` fixes the "outward direction" if the single-sided feature is enabled:
		
        if (m_front || edgeA->m_singleSided)   // added "OR singleSided" a couple of times

 2. `b2ChainShape::GetChildEdge()` copies the `bool m_singleSided` over to the target `b2Edge`.

What do you think? I am about to check out what I exactly did back then and reapply it to an up-to-date (git HEAD) Box2D and make a nice PR. Are you interested? Is the name `singleSided` fine for you?

**Comment by erincatto** [comment]
I think it is reasonable to have one-sided collision. However, I am concerned about giving up the centroid test. If you ever needed to have a thin wall in a level design, you could easily put to such edges back to back. This could lead to collision popping to the other side of the wall.

**Comment by erincatto** [comment]
One-sided edge collision was added in #607.
